src/content/reference/rules/never-call-component-functions-directly.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Never call component functions directly
+---
+
+<Intro>
+Components should only be used in JSX. Don't call them as regular functions.
+</Intro>
+
+---
+
+React takes care of _when_ your code runs for you so that your application has a great user experience. It is declarative: you tell React what to render in your component's logic, and React will figure out how best to display it to your user.
+
+In order to do this, React must decide when your component function is called during rendering. In React, you do this using JSX.
+
+```js {2}
+function BlogPost() {
+  return <Layout><Article /></Layout>; // ✅ Only use components in JSX
+}
+```
+
+```js {3}
+function BlogPost() {
+  ReactDOM.render(
+    Layout({ children: Article() }), // ❌ Never call them directly
+    /* ... */
+  );
```

Comment:
This is a deprecated API, but I think you meant something like this?

```suggestion
  return Layout({ children: Article() }); // ❌ Never call them directly
```

Reply:
Not sure if the particular pattern matters here but what I've seen most often in product code is the mix of JSX/functions, like:

```
function BlogPost() {
  return <Layout>{Article()}</Layout>
}
```